textworld/core.py

```diff
@@ -47,6 +48,12 @@ def __init__(self, **kwargs):
         #: bool: All the facts that are currently true about the world.
         #:       This information changes from one step to another.
         self.facts = kwargs.get("facts", False)
+        #: bool: Mutually exclusive sets of winning facts for each quest.
```

Comment:
Notice that win/fail_facts return a **Mutually exclusive sets** of winning/failing facts **for each quest**, i.e. `List[List[Tuple[Proposition]]]`.